driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverFactory.java

```diff
@@ -248,33 +247,35 @@ public void setupSession(EventFiringWebDriver eventFiringWebDriver, DesktopWebDr
         }
     }
 
-    private void setWindowSizeBasedOnDisplayResolution(WebDriver.Window window, String browser) {
-        log().debug("Trying to set window size to: " + Defaults.DISPLAY_RESOLUTION);
-        String[] split = Defaults.DISPLAY_RESOLUTION.split("x");
+    private void setWindowSize(WebDriver.Window window, Dimension dimension) {
+        window.setPosition(new Point(0, 0));
+        window.setSize(dimension);
+    }
+
+    private Dimension getTargetWindowSize() {
+        String windowSizeProperty = PropertyManager.getProperty(TesterraProperties.WINDOW_SIZE, PropertyManager.getProperty(TesterraProperties.DISPLAY_RESOLUTION, "1920x1080"));
+        String[] split = windowSizeProperty.split("x");
         int width = Integer.parseInt(split[0]);
```

Comment:
In case of non-digit chars a NumberFormatException occurs. What will happend with this exception? Is it catched at any other place?
`Integer.getInteger()` returns `null`, so you only have to check for NULL values.

Reply:
Improved parsing